File: educational-core/src/com/jetbrains/edu/learning/courseFormat/tasks/CodeTask.java
package com.jetbrains.edu.learning.courseFormat.tasks;

import com.jetbrains.edu.learning.courseFormat.CheckStatus;
import org.jetbrains.annotations.NotNull;

import java.util.Date;

public class CodeTask extends Task {
  @SuppressWarnings("unused") //used for deserialization
  public CodeTask() {}

  public CodeTask(@NotNull final String name, int id, int position, @NotNull Date updateDate, @NotNull CheckStatus status) {
    super(name, id, position, updateDate, status);
  }

  @Override
  public String getItemType() {
    return "code";
  }

  @Override
  public boolean supportSubmissions() {
    return true;
  }
}

File: educational-core/src/com/jetbrains/edu/learning/courseFormat/tasks/CodeTask.kt
package com.jetbrains.edu.learning.courseFormat.tasks

import com.jetbrains.edu.learning.courseFormat.CheckStatus
import java.util.*

class CodeTask : Task {
  //used for deserialization
  constructor()

  constructor(name: String) : super(name)

  constructor(name: String, id: Int, position: Int, updateDate: Date, status: CheckStatus) :
    super(name, id, position, updateDate, status)

  override fun getItemType(): String = CODE_TASK_TYPE

  override fun supportSubmissions(): Boolean = true

  override fun isPluginTaskType(): Boolean = false

  companion object {
    const val CODE_TASK_TYPE: String = "code"
  }
}